Add formulas for distributions

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69956
* #69884
* #69727
* #69740
* __->__ #69690
* #69661


Differential Revision: [D33031726](https://our.internmc.facebook.com/intern/diff/D33031726)

soulitzer committed:
Add formulas for distributions

[ghstack-poisoned]

albanD reviewed (approved):
We want to make these non-differentiable in the future. So sure.

soulitzer committed:
Update on "Add formulas for distributions"




[ghstack-poisoned]

soulitzer committed:
Update on "Add formulas for distributions"




[ghstack-poisoned]

albanD commented on tools/autograd/derivatives.yaml:
```diff
@@ -1822,15 +1823,19 @@
 
 - name: threshold(Tensor self, Scalar threshold, Scalar value) -> Tensor
   self: threshold_backward(grad, self, threshold)
+  result: auto_element_wise
```
why did this got moved here?

soulitzer replied:
faulty rebase, fixed now!

soulitzer committed:
Update on "Add formulas for distributions"




[ghstack-poisoned]

soulitzer committed:
Update on "Add formulas for distributions"




[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33030711).

soulitzer committed:
Update on "Add formulas for distributions"


Differential Revision: [D33030711](https://our.internmc.facebook.com/intern/diff/D33030711)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

albanD reviewed (approved):
Thanks for the cleanup of the rebase

soulitzer committed:
Update on "Add formulas for distributions"



Differential Revision: [D33031726](https://our.internmc.facebook.com/intern/diff/D33031726)

[ghstack-poisoned]

soulitzer committed:
Update on "Add formulas for distributions"



Differential Revision: [D33031726](https://our.internmc.facebook.com/intern/diff/D33031726)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33031726).

soulitzer committed:
Update on "Add formulas for distributions"



Differential Revision: [D33031726](https://our.internmc.facebook.com/intern/diff/D33031726)

[ghstack-poisoned]
